src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1519,4 +1519,19 @@
     <value>and all other platforms</value>
     <comment>This call site is reachable on: 'windows' 10.0.2000 and later, and all other platforms</comment>
   </data>
+  <data name="UseValidPlatformStringDescription" xml:space="preserve">
+    <value>Platform compatibility analyzer would not work properly if invalid platform name or version used.</value>
+  </data>
+  <data name="UseValidPlatformStringTitle" xml:space="preserve">
+    <value>Use valid platform string</value>
+  </data>
+  <data name="UseValidPlatformStringUnknownPlatform" xml:space="preserve">
+    <value>The platform '{0}' is not a known platform name</value>
+  </data>
+  <data name="UseValidPlatformStringInvalidVersion" xml:space="preserve">
+    <value>The '{0}' is not a valid version. Please provide an input having 2 to {1} numbers separated with a dot.</value>
+  </data>
```

Comment:
You'll want a message for when there are exactly 2 versions so that it doesn't show a range of "2 to 2"

Reply:
For the existing `OperatingSystem.Is<Platform>VersionAtLeast(...)` APIs there is no variant having 2 version parts, i am not sure if we will have a platform with a max of 2 version parts in the future, do you think that is possible for the future platforms?

Reply:
I'm not sure, but I'd rather not have to worry about updating this if one should emerge.

Reply:
I'm gonna use a resource string `Version '{0}' is not valid for platform '{1}'. Use a version with 2{2} parts for this platform.` where `3` or `4` part will be added with dash `-`, otherwise empty string which can express exact 2 versions, hope there is no localization issue with that cc @cristianosuzuki77